pkg/scheduler/framework/plugins/nodeaffinity/node_affinity.go

```diff
@@ -83,10 +90,54 @@ func (s *preFilterState) Clone() framework.StateData {
 // failed by this plugin schedulable.
 func (pl *NodeAffinity) EventsToRegister() []framework.ClusterEventWithHint {
 	return []framework.ClusterEventWithHint{
-		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add | framework.Update}},
+		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add | framework.Update}, QueueingHintFn: pl.isSchedulableAfterNodeChange},
 	}
 }
 
+// isSchedulableAfterNodeChange is invoked whenever a node changed. It checks whether
+// that change made a previously unschedulable pod schedulable. It errs on the side of
+// letting a pod scheduling attempt happen.
+func (pl *NodeAffinity) isSchedulableAfterNodeChange(pod *v1.Pod, oldObj, newObj interface{}) framework.QueueingHint {
+	if newObj == nil {
+		// Deletes don't make a pod schedulable.
+		return framework.QueueSkip
+	}
+
+	modifiedNode, ok := newObj.(*v1.Node)
```

Comment:
Please use As function to convert oldObj and newObj.
https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/util/utils.go#L167